packages/www/components/Sidebar/index.tsx

```diff
@@ -73,9 +74,16 @@ export type SidebarId =
 
 const Sidebar = ({ id }: { id: SidebarId }) => {
   const { user, logout } = useApi();
+
   const June = useJune();
 
-  June?.track(`sidebar ${id}`);
+  const trackSidebarEvent = useCallback(() => {
+    if (June) June.track(`sidebar ${id}`);
+  }, [June]);
+
+  useEffect(() => {
+    trackSidebarEvent();
```

Comment:
Could you explain what this useEffect is supposed to do? Is it on mount we fire this event?

Reply:
Yeah! just want to make sure the event is fired once the component is mounted. Do you think we can update with a better approach? I don't like using useEffect here as well :)

Reply:
Yes, the sidebar id in which the page is active, like `stream`, `asset`, `developers` is being tracked and sent to June

Reply:
Okay maybe we use pageview events for this?
https://www.june.so/docs/quickstart/install

Reply:
yeah, i've checked that, but they don't have that method on their SDK. We might need to add their API request then. 
Should I do it @0xcadams?

Reply:
Oh odd - maybe we just add a route change handler instead with the events (supported by their SDK)?

```
const router = useRouter(); 

  useEffect(() => {
    const handleRouteChange = (url, { shallow }) => {
      if (analytics) analytics.page(url);
    };

    router.events.on('routeChangeStart', handleRouteChange);

    return () => {
      router.events.off('routeChangeStart', handleRouteChange);
    };
  }, [analytics]);
  ```

Reply:
yep! sounds good! just updated it.

Reply:
LGTM!